src/constructs/ecs/ecs-task.ts

```diff
@@ -236,6 +236,20 @@ export class GuEcsTask extends Construct {
       taskDefinition.addToTaskRolePolicy(distPolicy);
     }
 
+    // See https://docs.aws.amazon.com/guardduty/latest/ug/prereq-runtime-monitoring-ecs-support.html#before-enable-runtime-monitoring-ecs
+    const guardDutyPolicy = new PolicyStatement({
+      effect: Effect.ALLOW,
+      actions: [
+        "ecr:GetAuthorizationToken",
+        "ecr:BatchCheckLayerAvailability",
+        "ecr:GetDownloadUrlForLayer",
+        "ecr:BatchGetImage",
+      ],
+      resources: ["*"],
```

Comment:
Does this need to be so permissive ?

Reply:
No it doesn't! I left it as * because there's a different guard duty fargate url for each region. I've updated it now to only allow fetching images from the eu-west-1 guard duty fargate container location, as referenced here https://docs.aws.amazon.com/guardduty/latest/ug/runtime-monitoring-ecr-repository-gdu-agent.html